### PR TITLE
add internal release notes

### PR DESCRIPTION
**Pull request by davidhewitt**
To accompany the 0.17.2 release, I thought I would document the release process so that we all know it and are aware of how it works, as it's also getting slightly more fiddly with the addition of `towncrier` :)

### CODE REVIEW

File: Releasing.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+## 3. Leave for a cooling off period
+
+Wait a couple of days in case anyone wants to hold up the release to add bugfixes etc.
+
+## 4. Put live
+
+To put live:
+- 1. run `nox -s publish` to put live on crates.io
+- 2. publish the release on Github
+- 3. merge the release PR
```
Comment (adamreichold) [inline Releasing.md:44]: I noticed that we have quite a branches like `release-<major>.<minor>.<patch>` lying around in the repository. Aren't those redundant to the `v<major>.<minor>.<patch>` tags and could be delete in this step? (In contrast to the `release-<major>.<minor>` branches for diverging maintenance between the current and the next release on `main`.)

File: Releasing.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+4. Create the commit containing all the above changes, with a message of `release: <version>`. Push to `release-<version>` branch on the main PyO3 repository.
+
+## 2. Create the release PR and draft release notes
+
+Open a PR for the branch, and confirm that it passes CI.
+
+On https://github.com/PyO3/pyo3/releases, click "Draft a new release". The tag will be a new tag of `v<version>` (note preceding `v`) and target should be the `release-<version>` branch you just pushed.
+
+Write release notes which match the style of previous releases. You can get the list of contributors by running `nox -s contributors -- v<prev-version> release-<version>` to get contributors from the previous version tag through to the branch tip you just pushed.
```
Comment (adamreichold) [inline Releasing.md:31]: Maybe it would be good to slightly reword this to indicate that the Nox session will only _after_ pushing to GitHub as it uses the GitHub API instead of the local repository state?

File: Releasing.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+   - Add _new_ lines to the netlify `_redirects` file rather than replacing the existing ones.
+   - Make sure not to modify the CHANGELOG during this step!
+
+2. Run `towncrier build` to generate the CHANGELOG. The version used by `towncrier` should automatically be correct because of the update to `pyproject.toml` in step 1.
+
+3. Manually edit the CHANGELOG for final notes. Steps to do:
+   - Adjust wording of any release lines to make them clearer for users / fix typos.
+   - Add a new link at the bottom for the new version, and update the `Unreleased` link.
+
+4. Create the commit containing all the above changes, with a message of `release: <version>`. Push to `release-<version>` branch on the main PyO3 repository.
```
Comment (messense) [inline Releasing.md:23]: We can add a check to the existing changelog CI to ensure that `newsfragrments` directory contains no news entries when PR title starts with `release: `.

File: Releasing.md (hunk 1 above)
Comment (davidhewitt) [reply Releasing.md:44]: Yes, good idea to clean these up.

I'd been keeping the maintenance branches of each version as 0.14, 0.15, 0.16, etc.

I don't have a strong opinion whether it's better to call the maintenance branch `0.14` or `release-0.14`, any option you prefer? I can also add instructions here to create the maintenance branch if pushing a new breaking release.

File: Releasing.md (hunk 2 above)
Comment (davidhewitt) [reply Releasing.md]: Good idea, done 👍

File: Releasing.md (hunk 1 above)
Comment (adamreichold) [reply Releasing.md:46]: > I don't have a strong opinion whether it's better to call the maintenance branch 0.14 or release-0.14,

I have no strong opinion but I would be glad if we use consistent names.

> I can also add instructions here to create the maintenance branch if pushing a new breaking release.

That would be nice. Thanks!

Comment (davidhewitt) [reply Releasing.md:46]: Let's go for `release-0.x`. That way the created branch from the PR can stick around as the maintenance branch when needed. I'll also tidy up the existing set of branches.

**Review by messense — APPROVED** [review]

File: Releasing.md (hunk 1 above)
Comment (davidhewitt) [reply Releasing.md:48]: Updated and cleaned up existing branches.
